generator/src/main/java/com/linkedin/pegasus/generator/ProjectionMaskApiChecker.java

```diff
@@ -0,0 +1,64 @@
+package com.linkedin.pegasus.generator;
+
+import com.linkedin.data.schema.DataSchemaLocation;
+import com.linkedin.pegasus.generator.spec.ClassTemplateSpec;
+import com.sun.codemodel.JClass;
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+/**
+ * Utility to check if a nested type has or will generate projection mask APIs.
+ */
+public class ProjectionMaskApiChecker
+{
+  private final TemplateSpecGenerator _specGenerator;
+  private final Set<String> _sourceFiles;
+  private final ClassLoader _classLoader;
+  private final Map<JClass, Boolean> _hasProjectionMaskCache = new HashMap<>();
+
+  ProjectionMaskApiChecker(TemplateSpecGenerator specGenerator,
+      Set<File> sourceFiles, ClassLoader classLoader)
+  {
+    _specGenerator = specGenerator;
+    _sourceFiles = sourceFiles.stream().map(File::getAbsolutePath).collect(Collectors.toSet());
+    _classLoader = classLoader;
+  }
+
+  /**
+   * Returns true if any of the conditions below is true.
+   * <ul>
+   *   <li>The passed in class can be loaded from class path and also contains the ProjectionMask class.</li>
+   *   <li>The passed in class will be generated from a source PDL file on which the template generator is running.</li>
+   * </ul>
+   */
+  boolean hasProjectionMaskApi(JClass definedClass, ClassTemplateSpec templateSpec)
+  {
+    return _hasProjectionMaskCache.computeIfAbsent(definedClass, (jClass) ->
+    {
+      try
+      {
+        final Class<?> clazz = _classLoader.loadClass(jClass.fullName());
+        return Arrays.stream(clazz.getClasses()).anyMatch(c -> c.getSimpleName().equals("ProjectionMask"));
```

Comment:
Should we use some regex? or we are pretty sure it will be named like `ProjectionMask`?

Reply:
`projectionMask`, `ProjectionMasks`?

Reply:
It will always be "ProjectionMask" as we generate it and control what it would be called. I'll make it a constant to make it explicit.